processor/k8sattributesprocessor/README.md

```diff
@@ -1,10 +1,15 @@
 # Kubernetes Attributes Processor
+<!-- status autogenerated section -->
+| Status                   |           |
+| ------------------------ |-----------|
+| Stability                | [beta]   |
+| Supported pipeline types | logs, metrics, traces   |
+| Distributions            | [contrib] |
+| Warnings                 | [Statefulness](#warnings) |
```

Comment:
I think the right place to send users would be https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/standard-warnings.md#statefulness, does that make sense here?

Reply:
Sure, how do you want to go about this?

Reply:
I think it is appropriate for the components to always have their own `warnings` section that can then link out to the standard warnings.  If a component needs a standard warning then it should also be able to clarify exactly how that warning applies.

For this processor, is it actually stateful?

Reply:
> For this processor, is it actually stateful?


Depends on what we call stateful. This processor keeps k8s API state, not any pdata state. If we take definition from https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/cumulativetodeltaprocessor#warnings, it's not stateful

Reply:
I would say it is not stateful; the deployment architecture of the collector is not affected by the k8sattributes processor (passthrough doesn't count because you don't have to use it).  It may keep extra information in memory, but the term Stateful implies you need to send all the related telemetry to the same collector instance, and that is not true for k8sattributes.

I think it could be fair to add a warning about memory consumption.  Since the processor has to remember the metadata of the node it is on, it consumes more memory than other processors.  That consumption is compounded if you don't filter down to only the metadata for the node it is on.  I think those are fair warnings to call out.

Reply:
That's a great call out! I agree with the additional _memory_ warning instead

Reply:
It feels like this needs to be kicked off to its own issue and separate PR. I can remove the warnings for now and get this in and we can hammer the wording separately.

Reply:
I'm going to remove the stateful warning and I have filed #21789 to follow up.